lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,325 @@
+use std::result::Result;
+use vrl::prelude::*;
+
+mod logfmt {
+    use std::collections::{BTreeMap, HashSet};
+    use std::fmt::Write;
+
+    use shared::btreemap;
+    use Value::{Array, Object};
+
+    use vrl::prelude::*;
+
+    fn encode_string(output: &mut String, str: &str) {
+        let needs_quotting = str.find(' ').is_some();
+        if needs_quotting {
+            output.write_char('"').unwrap();
+        }
+
+        for c in str.chars() {
+            let needs_escaping = matches!(c, '\\' | '"');
+            if needs_escaping {
+                output.write_char('\\').expect("");
+            }
+
+            output.write_char(c).unwrap();
+        }
+
+        if needs_quotting {
+            output.write_char('"').unwrap();
+        }
+    }
+
+    fn encode_value(output: &mut String, value: &Value) {
+        match value {
+            Value::Bytes(b) => {
+                let val = String::from_utf8_lossy(b);
+                encode_string(output, &val)
+            }
+            _ => {
+                let val = format!("{}", value);
+                encode_string(output, &val)
+            }
+        }
+    }
+
+    fn flatten(
+        input: &BTreeMap<String, Value>,
```

Comment:
This would also change to take ownership of `input` to avoid unneeded cloning below.

This could get a bit tricky, but I think would help improve performance here.